iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/CitationsActionBuilderModuleFactory.java

```diff
@@ -1,118 +1,203 @@
+        private ConfidenceLevelValidator confidenceLevelValidator = new ConfidenceLevelValidator();
+        private CitationEntryNormalizer citationEntryNormalizer = new CitationEntryNormalizer();
+        private BlobCitationEntryBuilder blobCitationEntryBuilder = new BlobCitationEntryBuilder();
+
+        public SortedSet<BlobCitationEntry> convert(List<CitationEntry> source, Float trustLevelThreshold) {
+            if (source != null) {
+                Float confidenceLevelThreshold = trustLevelConverter.convert(trustLevelThreshold);
+                return source.stream()
+                        .map(citationEntry -> confidenceLevelValidator.validate(citationEntry, confidenceLevelThreshold))
+                        .map(citationEntry -> citationEntryNormalizer.normalize(citationEntry))
+                        .map(citationEntry -> blobCitationEntryBuilder.build(citationEntry))
+                        .collect(Collectors.toCollection(TreeSet::new));
+            }
+            return null;
         }
 
-        // ------------------------ PRIVATE --------------------------
-        
         /**
-         * Performs confidence level normalization. Removes empty lists. 
-         * Removes 50| prefix from publication identifier.
-         * 
-         * @param source list of citations to be normalized
-         * @return {@link BlobCitationEntry} objects having confidence level value normalized
+         * Allows trust level threshold to be converted to confidence level threshold.
          */
-        private SortedSet<BlobCitationEntry> normalize(List<CitationEntry> source) {
-            if (source != null) {
-                SortedSet<BlobCitationEntry> results = new TreeSet<BlobCitationEntry>();
-                for (CitationEntry currentEntry : source) {
-                    if (currentEntry.getExternalDestinationDocumentIds().isEmpty()) {
-                        currentEntry.setExternalDestinationDocumentIds(null);
-                    }
-                    if (currentEntry.getDestinationDocumentId() != null) {
-                        currentEntry.setDestinationDocumentId(
-                                StringUtils.split(currentEntry.getDestinationDocumentId().toString(),
-                                        InfoSpaceConstants.ROW_PREFIX_SEPARATOR)[1]);
-                    }
-                    results.add(CitationsActionBuilderModuleUtils.build(currentEntry));
+        public static class TrustLevelConverter {
+            private Float conversionFactor;
+
+            public TrustLevelConverter(Float conversionFactor) {
+                this.conversionFactor = conversionFactor;
+            }
+
+            public Float convert(Float trustLevelThreshold) {
+                return Optional.ofNullable(trustLevelThreshold)
+                        .map(x -> x / conversionFactor)
+                        .orElse(null);
```

Comment:
IDE does not hint the usage of `Optional` here, it was my initial choice. Your code seems better because I'm unnecessary wrapping the value in an `Optional` and then unwrapping it upon returning. This pattern works better when we have more than one operation to apply to initial value because every operation is isolated. This was probably why I've chosen this solution, but in the future I'll use the simpler solution in situations like this.